I'm getting
```
File "src/sage/repl/ipython_kernel/install.py", line 253, in sage.repl.ipython_kernel.install.SageKernelSpec.check
Failed example:
    SageKernelSpec.check()  # random
Exception raised:
    Traceback (most recent call last):
      File "/Users/buildbot-sage/worker/sage_git/build/src/sage/doctest/forker.py", line 734, in _run
        self.compile_and_execute(example, compiler, test.globs)
      File "/Users/buildbot-sage/worker/sage_git/build/src/sage/doctest/forker.py", line 1158, in compile_and_execute
        exec(compiled, globs)
      File "<doctest sage.repl.ipython_kernel.install.SageKernelSpec.check[1]>", line 1, in <module>
        SageKernelSpec.check()  # random
        ^^^^^^^^^^^^^^^^^^^^^^
      File "/Users/buildbot-sage/worker/sage_git/build/src/sage/repl/ipython_kernel/install.py", line 267, in check
        if Path(shutil.which(spec.argv[0])).resolve() != Path(sys.executable).resolve():
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/Users/buildbot-sage/worker/sage_git/build/local/var/lib/sage/venv-python3.12.5/lib/python3.12/pathlib.py", line 1162, in __init__
        super().__init__(*args)
      File "/Users/buildbot-sage/worker/sage_git/build/local/var/lib/sage/venv-python3.12.5/lib/python3.12/pathlib.py", line 373, in __init__
        raise TypeError(
    TypeError: argument should be a str or an os.PathLike object where __fspath__ returns a str, not 'NoneType'
```
on some platforms, e.g. http://build.sagemath.org/#/builders/68/builds/6

--- a/src/sage/repl/ipython_kernel/install.py
+++ b/src/sage/repl/ipython_kernel/install.py
@@ -264,7 +264,12 @@ class SageKernelSpec:
             import sys
             import shutil
             from pathlib import Path
-            if Path(shutil.which(spec.argv[0])).resolve() != Path(sys.executable).resolve():
+            kernel_executable = shutil.which(spec.argv[0])
+            if not kernel_executable:
+                warnings.warn(f'The kernel named {ident} does not seem to be runnable; '
+                              'check your Jupyter configuration '
+                              '(see https://docs.jupyter.org/en/latest/use/jupyter-directories.html).')
+            if Path(kernel_executable).resolve() != Path(sys.executable).resolve():
                 warnings.warn(f'The kernel named {ident} does not seem to correspond to this '
                               'installation of SageMath; check your Jupyter configuration '
                               '(see https://docs.jupyter.org/en/latest/use/jupyter-directories.html).')

--- a/src/sage/repl/ipython_kernel/install.py
+++ b/src/sage/repl/ipython_kernel/install.py
@@ -269,6 +269,7 @@ class SageKernelSpec:
                 warnings.warn(f'The kernel named {ident} does not seem to be runnable; '
                               'check your Jupyter configuration '
                               '(see https://docs.jupyter.org/en/latest/use/jupyter-directories.html).')
+                return
             if Path(kernel_executable).resolve() != Path(sys.executable).resolve():
                 warnings.warn(f'The kernel named {ident} does not seem to correspond to this '
                               'installation of SageMath; check your Jupyter configuration '

